compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2078,7 +2078,8 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                             diag.span_suggestion(
                                 span,
                                 *msg,
-                                format!("{}.as_ref()", snippet),
+                                // issue# 100605, suggesting convert from &Option<T> to Option<&T>, remove the extra `&`
```

Comment:
Can you tag this with `// HACK:` or something? I would like to get rid of this eventually, but I am mostly ok with landing this now as a mitigation, since we do similar things other places (where we don't have access to HIR to get the real expressions)